### PR TITLE
Change the field mapping index time boost into a query time boost.

### PR DESCRIPTION
**Pull request by jimczi**
Index time boost will still be applied for indices created before 5.0.0.

### CODE REVIEW

**Comment by jimferenczi** [comment]
Originated from https://github.com/elastic/elasticsearch/issues/12394#issuecomment-190107719

File: core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java [hunk 1]
```diff
@@ -186,6 +186,7 @@ private static void reverseOrder(ParseContext.InternalParseContext context) {
         }
     }
 
+    // TODO Should we remove this feature ? It seems that it's not tested nor used.
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java]: +1 it looks like there are some remainders of the boost field mapper that we removed a while ago (eg. there is also ParseContext.setDocBoost).

File: docs/reference/mapping/params/boost.asciidoc [hunk 2]
```diff
@@ -1,8 +1,8 @@
-[[index-boost]]
+[[mapping-boost]]
```
Comment (jpountz) [inline docs/reference/mapping/params/boost.asciidoc]: Since you are removing an existing link, I think you need to add an entry to redirects.asciidoc?

Comment (jpountz) [inline docs/reference/mapping/params/boost.asciidoc]: @clintongormley could you confirm?

**Comment by jpountz** [comment]
I think the doc changes might break some links (@clintongormley should be able to confirm) but otherwise it looks good to me. Thanks!

File: docs/reference/mapping/params/boost.asciidoc (hunk 2 above)
Comment (clintongormley) [inline docs/reference/mapping/params/boost.asciidoc]: Yeah, adding to redirects would be good.

**Comment by jimferenczi** [comment]
@jpountz I've removed the docBoost from the context and added an entry in the redirects.asciidoc.
Could you take a look ?

**Comment by jpountz** [comment]
This looks great!

**Pull request merged**